tests/test_AggregateConnectionData.cpp

```diff
@@ -0,0 +1,719 @@
+ 'PROD' 5 5 2 2   3*  0.2   3*  'X' /
+
+'WINJ' 10 1  9 9   3*  0.2   3*  'X' /
+ 'WINJ'   9 1  9 9   3*  0.2   3*  'X' /
+ 'WINJ'   8 1  9 9   3*  0.2   3*  'X' /
+ 'WINJ'   7 1  9 9   3*  0.2   3*  'X' /
+ 'WINJ'   6 1  9 9   3*  0.2   3*  'X' /
+/
+
+WELSEGS
+
+-- Name    Dep 1   Tlen 1  Vol 1
+  'PROD'   7010      10    0.31   'INC' /
+
+-- First   Last   Branch   Outlet  Length   Depth  Diam  Ruff  Area  Vol
+-- Seg     Seg    Num      Seg              Chang
+-- Main Stem
+    2       12     1        1         20     20    0.2   1.E-3  1*   1* /
+-- Top Branch
+    13      13     2        3         50      0    0.2   1.E-3  1*   1* /
+    14      17     2        13       100      0    0.2   1.E-3  1*   1* /
+ /
+
+COMPSEGS
+
+-- Name
+  'PROD' /
+
+-- I  J  K  Brn  Start   End     Dirn   End
+--          No   Length  Length  Penet  Range
+-- Top Branch
+  1  5  2  2         30      130     'X'    3* /
+  2  5  2  2        130      230     'X'    3* /
+  3  5  2  2        230      330     'X'    3* /
+  4  5  2  2        330      430     'X'    3* /
+  5  5  2  2        430      530     'X'    3* /
+-- Middle Branch
+ /
+
+
+WCONPROD
+ 'PROD' 'OPEN' 'LRAT'  3*  2000  1*  2500  1*  /
+ /
+
+WCONINJE
+ 'WINJ' 'WAT' 'OPEN' 'BHP'  1*  20000  3500  1*  /
+ /
+
+
+TUNING
+ /
+ /
+ /
+
+TSTEP
+ 2 2
+/
+
+
+END
+
+
+)~" };
+
+        return Opm::Parser{}.parseString(input);
+    }
+
+    
+    Opm::data::WellRates wr()
+    {
+        using o = ::Opm::data::Rates::opt;
+
+        auto xw = ::Opm::data::WellRates{};
+
+        {
+            xw["PROD"].rates
+                .set(o::wat, 1.0)
+                .set(o::oil, 2.0)
+                .set(o::gas, 3.0);
+	    xw["PROD"].bhp = 213.0;
+	    double qo = 5.;
+	    double qw = 4.;
+	    double qg = 50.;
+            for (int i = 0; i < 5; i++) {
+		xw["PROD"].connections.emplace_back();
+		auto& c = xw["PROD"].connections.back();
+		
+		c.rates.set(o::wat, qw*(float(i)+1.))
+                   .set(o::oil, qo*(float(i)+1.))
+                   .set(o::gas, qg*(float(i)+1.));
+		c.pressure = 215.; 
+	    }
+	    auto seg = Opm::data::Segment{};
+	    for (std::size_t i = 1; i < 5; i++) {
+		xw["PROD"].segments.insert(std::pair<std::size_t,Opm::data::Segment>(i,seg));
+	    }
+            xw["WINJ"].bhp = 234.0;
+
+            xw["WINJ"].rates.set(o::wat, 5.0);
+	    xw["WINJ"].rates.set(o::oil, 0.0);
+	    xw["WINJ"].rates.set(o::gas, 0.0);
+	    qw = 7.;
+            for (int i = 0; i < 5; i++) {
+		xw["WINJ"].connections.emplace_back();
+		auto& c = xw["WINJ"].connections.back();
+		
+		c.rates.set(o::wat, qw*(float(i)+1.))
+		       .set(o::oil, 0.)
+		       .set(o::gas, 0.);
+		c.pressure = 218.;
+	    }
+        }
+        return xw;
+    }
+}
+
+struct SimulationCase
+{
+    explicit SimulationCase(const Opm::Deck& deck)
+        : es   ( deck )
+        , grid(deck)
+        , sched( deck, es )
+    {}
+
+    // Order requirement: 'es' must be declared/initialised before 'sched'.
```

Comment:
Noise comment.

Reply:
> Noise comment.

Not really.  It alerts a developer making changes here that the order in which these members are declared actually matters.  Since the `sched` member depends on the `es` member, `sched` really must be declared after `es`.

Reply:
Well - the compiler will tell if you get it wrong.

Reply:
> Well - the compiler will tell if you get it wrong.

That is certainly true.  On the other hand, the compiler (probably, I haven't tested) won't complain if you write
```C++
struct SimulationCase
{
    explicit SimulationCase(const Opm::Deck& deck)
        : sched(deck, es)
        , es   (deck)
    {}
    Opm::Schedule     sched;
    Opm::EclipseState es;
};
```
If the types were not `Schedule` and `EclipseState` this **might** be kosher (e.g., if the `sched` object internally just stores a reference to `es` and doesn't read from it until any of its member functions are called).  In the context of `Schedule` and `EclipseState` this is of course invalid.